### ISSUE TITLE
implement report_after_job_retries in ActiveJob adapter

### ISSUE THREAD

**Issue by modosc** [getsentry/sentry-ruby#2499]
we are using `sentry-rails` to report errors from `ActiveJob` (which is running on `sidekiq`). we want the ability to only report on errors when retries are exhausted. there doesn't seem to be a way to do this directly in the `ActiveJob` integration (`sentry-sidekiq`, `sentry-delayed_job`, and `sentry-resque` all support this configuration option).

is it possible to implement this directly in the `ActiveJob` adapter?

**Comment by sl0thentr0py** [comment]
Yes, we should implement  `:report_after_job_retries` for active job too, feel free to make a PR if you want it quicker!

**Comment by modosc** [comment]
@sl0thentr0py i gave it a shot in #2500 but ran into several issues:
1. tests work locally but not in CI or in CI but not locally, similar to https://github.com/getsentry/sentry-ruby/pull/2487 
2. rails 5.0 didn't support `retry_on` and i'm unclear on how you expect that to be handled (print a deprecation warning? throw an error?). 

anyway - i'm closing the pr out but i'll make sure the branch still exists. i think the approach is valid (and it seems to work) but i can't spend anymore time figuring out the tests.

**Comment by sl0thentr0py** [comment]
@modosc thanks a lot! I will reopen the PR and get it into a mergeable state myself. You don't have to do further work on it.

**Comment by sl0thentr0py** [comment]
@modosc sorry I'm juggling a few tasks since the year just started. I'll try to set aside some time on Friday for this.

**Comment by modosc** [comment]
> [@modosc](https://github.com/modosc) sorry I'm juggling a few tasks since the year just started. I'll try to set aside some time on Friday for this.

thanks - we actually came up with a temporary workaround so it's not as urgent as it was last week. 

also i rebased / updated my branch and there are fewer failures (but still many failing checks).

**Comment by solnic** [comment]
This was done via #2573
